bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/core/GraphqlProductViewHandler.java

```diff
@@ -151,23 +156,40 @@ private static void addProductConstraint(PredicateGroup gqlPredicateGroup) {
     protected ViewQuery createQuery(SlingHttpServletRequest request,
         Session session, String queryString) throws RepositoryException {
 
-        PredicateGroup gqlPredicateGroup = new PredicateGroup();
         queryString = preserveWildcards(queryString);
 
-        gqlPredicateGroup = PredicateConverter.createPredicatesFromGQL(queryString);
-        tagManager = request.getResourceResolver().adaptTo(TagManager.class);
+        PredicateGroup gqlPredicateGroup = PredicateConverter.createPredicatesFromGQL(queryString);
+        ResourceResolver resolver = request.getResourceResolver();
+        tagManager = resolver.adaptTo(TagManager.class);
         Set<String> predicateSet = customizePredicateGroup(gqlPredicateGroup);
 
         // set default start path
         RequestPathInfo pathInfo = request.getRequestPathInfo();
-        final CommerceBasePathsService cbps = request.getResourceResolver().adaptTo(CommerceBasePathsService.class);
+        CommerceBasePathsService cbps = resolver.adaptTo(CommerceBasePathsService.class);
         String defaultStartPath = cbps.getProductsBasePath();
         String startPath = (pathInfo.getSuffix() != null && pathInfo.getSuffix().startsWith(defaultStartPath)) ? pathInfo.getSuffix()
             : defaultStartPath;
         if (!predicateSet.contains(PathPredicateEvaluator.PATH)) {
             gqlPredicateGroup.add(new Predicate(PathPredicateEvaluator.PATH).set(PathPredicateEvaluator.PATH, startPath));
         }
 
+        String refererHeader = request.getHeader(REFERER_HEADER);
+        if (StringUtils.isNotBlank(refererHeader)) {
+            int p = refererHeader.lastIndexOf(PAGE_EDITOR_PATH);
+            String pagePath = refererHeader.substring(p + PAGE_EDITOR_PATH.length());
```

Comment:
You have to also check if `p` is positive, otherwise you end up doing `pagePath = refererHeader.substring(10)` and this could match something that doesn't have `editor.html` in the path.

Reply:
Hm, it's a bit more complicated than that but good catch. I'll fix it.